assets/training/model_management/src/azureml/model/mgmt/processors/transformers/convertors.py

```diff
@@ -187,6 +174,84 @@ def _save(
         metadata = fetch_mlflow_acft_metadata(base_model_name=self._model_id,
                                               is_finetuned_model=False,
                                               base_model_task=self._task)
+        if self._model_flavor == "OSS":
+            try:
+                self._save_in_oss_flavor(model, code_paths, input_example, pip_requirements, metadata)
+            except Exception as e:
+                logger.error("Model save failed with mlflow OSS flow for task: {} "
+                             "with exception: {}".format(self._task, e))
+
+                self._save_in_hftransformersv2_flavor(metadata, conda_env,
+                                                      code_paths, input_example,
+                                                      requirements_file, pip_requirements)
+        else:
+            self._save_in_hftransformersv2_flavor(metadata, conda_env,
+                                                  code_paths, input_example,
+                                                  requirements_file, pip_requirements)
+
+        # pin pycocotools==2.0.4
+        self._update_conda_dependencies({"pycocotools": "2.0.4"})
+
+        if self._task == SupportedTasks.AUTOMATIC_SPEECH_RECOGNITION.value:
+            self._update_conda_dependencies({"ffmpeg": "4.2.2"})
+
+    def _save_in_oss_flavor(self, model, code_paths, input_example, pip_requirements, metadata):
+        # create a conda environment for OSS transformers Flavor
+        python_version = platform.python_version()
+        pip_pkgs = self._get_curated_environment_pip_package_list()
+        conda_deps = CondaDependencies.create(conda_packages=None,
```

Comment:
from what I understand, the reason for creating new conda env here with specific dependencies is to load the model for saving to oss first and then persisting dependencies in conda.yaml. But I see this env being created, and model still being loaded into current component env (model-mgmt). If this is working in all case, would it make sense to not create conda env at all here and just provide dict as part of model save to oss?

Reply:
I am also wondering how loading the model in current env works but when conda is inferred by default for oss it adds some internal dependencies which break future valdn. 

Was this verified again recently if that is still the case?